### DISCUSSION TITLE
Select render custom selected value component

### DISCUSSION THREAD

**Discussion by kaisiant** [mantinedev/mantine#4144] [Q&A]
![image](https://user-images.githubusercontent.com/26655111/234520332-29af2e0a-5bb0-416e-8923-669ada47eb1a.png)
 
Hello, I want to render a custom selected component, basically just like the item component, with an icon and some text. Is it possible to do that? 
I don't think using inputContainer is the correct way, as it does not have the props like current selected value, handler to open the dropdown etc.

**Comment by rtivital** [comment]
No, it is not supported. You can use `icon` prop to display icon.

**Comment by needim** [comment]
Hi, it is actually a good feature. 

We had to write our custom select component which accepts custom `TriggerComponent` because Mantine select doesn't provide an API.

It is useful for designing page filters like Linear or similar apps.

<img width="395" alt="image" src="https://user-images.githubusercontent.com/271912/235449367-c62baa46-7f2d-4176-9ed9-312d61aa4dd3.png">
